virtcontainers/factory/factory.go

```diff
@@ -34,7 +34,8 @@ func (f *Config) validate() error {
 }
 
 type factory struct {
-	base base.FactoryBase
+	base             base.FactoryBase
+	hypervisorConfig vc.HypervisorConfig
```

Comment:
@devimc Please do not assume that there is only one user of a vm factory at one time. A VM's config is a property of the vm, not the factory. It is the caller's responsibility to call `InitializeVM` with a `hypervisorconfig` matching with the VM to be initialized. Or, you can add a `VMConfig` filed to struct VM in vm.go to record the config of a created vm.